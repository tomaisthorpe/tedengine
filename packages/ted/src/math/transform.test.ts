import TTransform from './transform';
import { quat, vec3 } from 'gl-matrix';

test('scale should default to (1,1,1)', () => {
  const t = new TTransform();
  expect([...t.scale]).toEqual([1, 1, 1]);
});

test('getMatrix should output expected matrix', () => {
  const t = new TTransform(
    vec3.fromValues(1, 2, 3),
    quat.fromEuler(quat.create(), Math.PI, 0.5 * Math.PI, 2 * Math.PI),
    vec3.fromValues(0.5, 2, 4)
  );

  expect(t.getMatrix()).toMatchSnapshot();
});
