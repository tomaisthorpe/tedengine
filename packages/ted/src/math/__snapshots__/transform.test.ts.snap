// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMatrix should output expected matrix 1`] = `
Float32Array [
  0.49680978059768677,
  0.054700739681720734,
  -0.013706065714359283,
  0,
  -0.21556973457336426,
  1.9853273630142212,
  0.10956613719463348,
  0,
  0.13281750679016113,
  -0.2059156596660614,
  3.992487668991089,
  0,
  1,
  2,
  3,
  1,
]
`;
